go-controller/pkg/util/util.go

```diff
@@ -31,3 +32,8 @@ func FetchIfMacWindows(interfaceName string) (string, error) {
 
 	return strings.ToLower(macAddress), nil
 }
+
+// IsServiceIPSet checks if the service is an headless service or not
```

Comment:
How is the name IPset related to "headless service"?

Reply:
It should be read as Is_ServcieIP_Set? This function already exists in the code base today. We just moved it under util.go to reuse it in several places.